File: _codep/ruby/best_buy/profit.rb
Profit = Class.new do
    def initialize(prices)
        @prices = prices
    end

    def get_max_profit
        max_profit = 0
        min_price = -1
        @prices.each do |p|
            if p < min_price || min_price == -1
                min_price = p
            elsif p - min_price > max_profit
                max_profit = p - min_price
            end
        end
        return max_profit
    end
end

File: _codep/java/best_buy/Profit.java
class Profit {
    private int[] mPrices;

    public Profit(int[] prices) {
         mPrices = prices;
    }

    public int getMaxProfit() {
        int maxProfit = 0;
        int minPrice = Integer.MAX_VALUE;
        for (int i = 0; i < mPrices.length; i++) {
            if (mPrices[i] < minPrice) {
                minPrice = mPrices[i];
            } else if (mPrices[i] - minPrice > maxProfit) {
                maxProfit = mPrices[i] - minPrice;
            }
        }
        return maxProfit;
    }
}
